src/app/ticker/actions.js

```diff
@@ -13,24 +13,57 @@ export const expiryLimit = () => async (dispatch: Function) =>
 
 export const RESERVED = 'ticker/RESERVED'
 
-export const reserve = () => async (dispatch: Function, getState: GetState) => {
+export const reserve = (polyCost: number) => async (dispatch: Function, getState: GetState) => {
   // TODO @bshevchenko: see below... const { isEmailConfirmed } = getState().pui.account
   const details: SymbolDetails = getState().form[formName].values
   const isInsufficientBalance = getState().pui.account.balance.lt(await TickerRegistry.registrationFee())
   dispatch(ui.tx(
-    [...(isInsufficientBalance ? ['Requesting POLY'] : []), 'Approving POLY spend', 'Token symbol reservation'],
+    [...(isInsufficientBalance ? ['Requesting POLY'] : []),
+      'Token Symbol Reservation Fee', 'Token Symbol Reservation'],
     async () => {
       if (isInsufficientBalance) {
         await PolyToken.getTokens(2500000)
       }
-      await TickerRegistry.registerTicker(details)
+      try {
+        await TickerRegistry.registerTicker(details)
+        dispatch(ui.notify(
+          'Spent ' + polyCost + ' POLY',
+          true
+        ))
+      } catch (e) {
+        throw e
+      }
     },
-    'Your Token Symbol Was Reserved Successfully',
+    'Your Token Symbol: ' + details.ticker + ', Was Reserved Successfully',
     () => {
-      dispatch({ type: RESERVED })
     },
     `/dashboard/${details.ticker}/providers`,
     undefined,
+    true, // TODO @bshevchenko: !isEmailConfirmed,
+    details.ticker.toUpperCase() + ' Token Symbol Reservation'
+  ))
+
+}
+
+export const faucet = (address: ?string, POLYamount: number) => async (dispatch: Function) => {
+  dispatch(ui.tx(
+    ['Receiving POLY From Faucet'],
+    async () => {
+      try {
+        await PolyToken.getTokens(25000, address)
+        dispatch(ui.notify(
+          'Received ' + POLYamount + ' POLY',
+          true
+        ))
+        dispatch(ui.setBalance(await PolyToken.myBalance()))
```

Comment:
I still can't change this because I can't get the callback in subscribeMyTransfers to fire when receiving POLY via the faucet. I looked at POLYtoken's events on etherscan, and I can't see a transfer event for getTokens(). If this was added, it hasn't been deployed.

Reply:
However, I can delete this line of code if you're confident it will work with the transfer event registration.